assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectZonesPage.tsx

```diff
@@ -0,0 +1,541 @@
+              </div>
+            </div>
+            {selectedRouteFilter !== "Bus" && (
+              <div className="mass-select-button-container">
+                <Button
+                  className={massSelectClassName(isAllSelected)}
+                  onClick={() =>
+                    selectGroups([LEFT_ZONES, MIDDLE_ZONES, RIGHT_ZONES])
+                  }
+                >
+                  {massSelectButtonLabels.left}
+                </Button>
+                <Button
+                  className={massSelectClassName(isLeftSelected)}
+                  onClick={() => selectGroups([LEFT_ZONES, MIDDLE_ZONES])}
+                >
+                  {massSelectButtonLabels.middle}
+                </Button>
+                <Button
+                  className={massSelectClassName(isRightSelected)}
+                  onClick={() => selectGroups([MIDDLE_ZONES, RIGHT_ZONES])}
+                >
+                  {massSelectButtonLabels.right}
+                </Button>
+              </div>
+            )}
+          </div>
+          <table className="zones-table">
+            <thead>
+              <tr className="table-header">
+                <th></th>
+                <th>All</th>
+                <th>{directionLabels.left}</th>
+                <th>{directionLabels.middle}</th>
+                <th>{directionLabels.right}</th>
+              </tr>
+            </thead>
+            <tbody>
+              {sortByStationOrder(filteredPlaces, selectedRouteFilter).map(
+                (place) => {
+                  return (
+                    <Fragment key={`place-zones-row-${place.id}`}>
+                      <PlaceZonesRow
+                        place={place}
+                        allSelectedSigns={value}
+                        setSignIds={onChange}
+                        route={selectedRouteFilter}
+                        routeToRouteIDMap={routeToRouteIDMap}
+                      />
+                      {hasRailFilter() && (
+                        <PlaceZonesRow
+                          place={place}
+                          allSelectedSigns={value}
+                          setSignIds={onChange}
+                          route={"Bus"}
+                          routeToRouteIDMap={routeToRouteIDMap}
+                        />
+                      )}
+                    </Fragment>
+                  );
+                },
+              )}
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const PlaceZonesRow = ({
+  place,
+  allSelectedSigns,
+  setSignIds,
+  route,
+  routeToRouteIDMap,
+}: {
+  place: Place;
+  allSelectedSigns: string[];
+  setSignIds: (signs: string[]) => void;
+  route: string;
+  routeToRouteIDMap: { [key: string]: string[] };
+}) => {
+  const allSignsForRouteAtPlace = place.screens.filter((screen) =>
+    fp.any(
+      (r) => routeToRouteIDMap[route]?.some((a) => r.startsWith(a)),
+      screen.route_ids,
+    ),
+  );
```

Comment:
**quibble:** it feels strange to me that this component is calculating these values instead of being passed them. i feel like this component should be passed lists of left, middle, and right zones maybe?